src/templates/TemplateData.ts

```diff
@@ -202,16 +231,56 @@ function getRuntimeKey(baseKey: string, runtime: ProjectRuntime): string {
     return runtime === ProjectRuntime.one ? baseKey : `${baseKey}.${runtime}`;
 }
 
-async function requestFunctionPortal<T>(hostname: string, subpath: string, runtime: string, param?: string): Promise<T> {
-    const options: request.OptionsWithUri = {
+async function cachedTemplatesAreCurrent(globalState?: vscode.Memento): Promise<boolean> {
+    if (globalState) {
+        for (const key of Object.keys(ProjectRuntime)) {
+            const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
+            const funcJsonOptions: request.OptionsWithUri = {
+                method: 'GET',
+                uri: funcCliFeedUrl,
+                headers: {
+                    'User-Agent': 'Mozilla/5.0' // Required otherwise we get Unauthorized
+                }
+            };
+
+            const cliFeedJson: cliFeedJsonResponse = <cliFeedJsonResponse>JSON.parse(await <Thenable<string>>request(funcJsonOptions).promise());
+            const currentRelease: string = <string>cliFeedJson.tags[runtime].release;
+            const cachedRelease: string | undefined = globalState.get(`${runtime}-release`);
+            if (!cachedRelease || parseInt(cachedRelease, 10) < parseInt(currentRelease, 10)) {
```

Comment:
Use the semver module for comparing versions. There's a few examples in this repo already